stan/math/prim/scal/fun/falling_factorial.hpp

```diff
@@ -51,9 +55,12 @@ namespace stan {
      */
     template<typename T1, typename T2>
     inline typename boost::math::tools::promote_args<T1, T2>::type
-    falling_factorial(const T1 x, const T2 n) {
-      using std::exp;
-      return exp(log_falling_factorial(x, n));
+    falling_factorial(const T1& x, const T2& n) {
+      if (is_nan(x) || is_nan(n))
```

Comment:
If you're going to restrict `n` to be an integer, you should do it here, too, for consistency. Then there's no NaN check required.